plugins/source/gcp/client/client.go

```diff
@@ -46,6 +50,12 @@ func (c *Client) Logger() *zerolog.Logger {
 }
 
 func New(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.ClientMeta, error) {
+	// We use a timeout to make sure that this method doesn't get stuck indefinitely.
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
```

Comment:
I think a minute is quite low. For example we saw users with more than 80k projects. 
Also I think there is no need for calling `cancel`

Reply:
Removed. Will be added at the SDK level if required later.